miden/src/benches/merkle_path_rescue_prime.rs

```diff
@@ -0,0 +1,212 @@
+use miden::{Assembler, Program, ProgramInputs, ProgramOutputs, ProofOptions, AdviceSet};
+use miden_core::{Felt, FieldElement, StarkField, Word};
+use rustbench::Benchmark;
+
+
+/// Pick a Merkle path of some depth (e.g., something like 32) and then  
+/// the job_size is the number of Merkle paths we verify. 
+/// So, for job_size=10 we verify 10 Merkle paths of depth 32, 
+/// for job_size=100 we verify 100 Merkle paths of depth 32 etc.
+
+pub struct Job {
+    num_iter: u32,
+    program: Program,
+    program_inputs: ProgramInputs,
+    proof_options: ProofOptions,
+    program_outputs: ProgramOutputs,
+    root_as_u64: Vec<u64>,
+}
+
+pub fn new_jobs() -> Vec<<Job as Benchmark>::Spec> {
+    vec![1, 10, 100, 1000]
+}
+
+impl Benchmark for Job {
+    const NAME: &'static str = "merkle_rescue_prime";
+    type Spec = u32;
+    type ComputeOut = Vec<u64>;
+    type ProofType = miden_prover::StarkProof;
+
+    fn job_size(spec: &Self::Spec) -> u32 {
+        *spec
+    }
+
+    fn output_size_bytes(output: &Self::ComputeOut, _proof: &Self::ProofType) -> u32 {
+        (output.len() * 8) as u32
+    }
+
+    fn proof_size_bytes(proof: &Self::ProofType) -> u32 {
+        proof.to_bytes().len() as u32
+    }
+    
+    fn new(num_iter: Self::Spec) -> Self {
+        let source = format!(
+            "  
+            begin
+            # verify a merkle path
+                repeat.{}
+      
+                    # stack = [d, i, R, ...]
+                    # duplicate the index and use as counter
+                    dup.1 push.1 add movdn.6 
+                    # stack = [d, i, R, (i+1) ...]
+        
+                    # verify merkle path for node i
+                    mtree_get
+                    # [V, R, (i+1) ...]
+
+                    # now we check if V is what we expect.
+                    # our leafs look like this: [0, 0, 0, counter]
+                    # get the word with the counter copied back
+                    dup.8 push.1 sub 
+                    #[i, V, R, (i+1),  ...]
+                    
+                    # then make it a Word
+                    padw drop
+                    #[0, 0, 0, i, V, R, (i+1),  ...]
+
+
+                    # if the two top words are equal, 
+                    # [0, 0, 0, i] == V,
+                    # we have the expected leaf
+                    eqw
+
+                    # fails if top stack element is 0
+                    inv 
```

Comment:
Interesting ! Can we also use `assert` ?

Reply:
Sure that would have been the better choice I guess